pkg/generator.go

```diff
@@ -195,15 +196,22 @@ func (g *Generator) getLocalizedPath(ctx context.Context, path string) string {
 	return toReturn
 }
 
-func upperFirstOnly(s string) string {
-	first := true
-	return strings.Map(func(r rune) rune {
-		if first {
-			first = false
-			return unicode.ToUpper(r)
-		}
-		return r
-	}, s)
+func (g *Generator) maybeMakeNameExported(name string, export bool) string {
+	if export && !ast.IsExported(name) {
+		return g.makeNameExported(name)
+	}
+
+	return name
+}
+
+func (g *Generator) makeNameExported(name string) string {
+	r, n := utf8.DecodeRuneInString(name)
+
+	if unicode.IsUpper(r) {
+		return name
+	}
+
+	return string(unicode.ToUpper(r)) + name[n:]
```

Comment:
by the way I made some quick benchmarks and this seems to be the best way to do this...it is much faster than `string -> []rune -> string` conversions